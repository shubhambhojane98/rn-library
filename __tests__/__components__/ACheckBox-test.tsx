import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import ACheckBox from '../../src/components/ACheckbox/ACheckbox';

it('renders correctly default prop', () => {
  const tree = renderer
    .create(
      <ACheckBox status="unchecked" label={'Checkbox'} onPress={jest.fn} />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with disabled prop', () => {
  const tree = renderer
    .create(
      <ACheckBox
        status="checked"
        label={'Checkbox'}
        onPress={jest.fn}
        disable={true}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
