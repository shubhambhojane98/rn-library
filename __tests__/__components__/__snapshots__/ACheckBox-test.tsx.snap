// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly default prop 1`] = `
<View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <SvgMock
        name="checkboxempty"
      />
      <Text
        style={
          Array [
            Object {
              "fontFamily": "Mulish-Regular",
              "fontSize": 21.13142857142857,
              "fontWeight": "normal",
            },
            Object {
              "color": "#000",
            },
            Object {
              "textAlign": "left",
            },
            Object {
              "fontSize": 19.65714285714286,
              "marginHorizontal": 2.4571428571428573,
            },
          ]
        }
        theme={
          Object {
            "colors": Object {
              "background": "#FFF",
              "backgroundLayout": "rgba(0, 0, 0, 0.8)",
              "black": "#000",
              "error": "#B00020",
              "greyOpac10": "rgba(53, 53, 53, 0.1)",
              "greyOpac50": "rgba(53, 53, 53, 0.5)",
              "lightgrey": "#D7D7D7",
              "linkColor": "#0000EE",
              "primary": "#FAD000",
              "primaryVariant": "#3700B3",
              "secondary": "#03DAC6",
              "secondaryVariant": "#018786",
              "shadowColor": "#222",
              "textColor": "#000",
              "transparent": "transparent",
              "white": "#FFF",
            },
            "dark": false,
          }
        }
      >
        Checkbox
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders correctly with disabled prop 1`] = `
<View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "opacity": 0.5,
        }
      }
    >
      <SvgMock
        name="checkboxfilled"
      />
      <Text
        style={
          Array [
            Object {
              "fontFamily": "Mulish-Regular",
              "fontSize": 21.13142857142857,
              "fontWeight": "normal",
            },
            Object {
              "color": "#000",
            },
            Object {
              "textAlign": "left",
            },
            Object {
              "fontSize": 19.65714285714286,
              "marginHorizontal": 2.4571428571428573,
            },
          ]
        }
        theme={
          Object {
            "colors": Object {
              "background": "#FFF",
              "backgroundLayout": "rgba(0, 0, 0, 0.8)",
              "black": "#000",
              "error": "#B00020",
              "greyOpac10": "rgba(53, 53, 53, 0.1)",
              "greyOpac50": "rgba(53, 53, 53, 0.5)",
              "lightgrey": "#D7D7D7",
              "linkColor": "#0000EE",
              "primary": "#FAD000",
              "primaryVariant": "#3700B3",
              "secondary": "#03DAC6",
              "secondaryVariant": "#018786",
              "shadowColor": "#222",
              "textColor": "#000",
              "transparent": "transparent",
              "white": "#FFF",
            },
            "dark": false,
          }
        }
      >
        Checkbox
      </Text>
    </View>
  </View>
</View>
`;
